pkg/systemd/generate/common.go

```diff
@@ -86,6 +86,32 @@ func filterCommonContainerFlags(command []string, argCount int) []string {
 	return processed
 }
 
+func escapeJSONString(s string) string {
+	if strings.HasPrefix(s, "[") && strings.HasSuffix(s, "]") {
+		return "'" + s + "'"
```

Comment:
my first question here is, what if `s` includes singlequotes? But I can't figure out how to trigger this code, so I can't write a test case, and I need to move on to other stuff. Anyhow, maybe you could add that to your unit tests? Something like:
```
... --entrypoint "[\"sh\", \"-c\", \"date '+%s'\"]"
```